File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
    int size = cost.size();
    vector<int> res(size);
    res[0] = cost[0];
    res[1] = cost[1];
    for (int i = 2;i < size;i++)
        res[i] = min(res[i - 1], res[i - 2]) + cost[i];
    return min(res[size - 1], res[size - 2]);


}
};

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        stack = []
        water = 0
        for i in range(len(height)):
            while stack and height[i] > height[stack[-1]]:
                bottom = stack.pop()
                if not stack:
                    break
                left_boundary = stack[-1]
                right_boundary = i
                width = right_boundary - left_boundary - 1
                bounded_height = min(height[left_boundary], height[right_boundary]) -  height[bottom]
                water += width * bounded_height
            stack.append(i)
        return water